writer/writer.ts

```diff
@@ -0,0 +1,72 @@
+import { Chroma } from "@langchain/community/vectorstores/chroma";
+import { getCourseDataForChroma, type ChromaDocumentData } from "../fetcher/course-document-fetcher";
+import type { Document } from "@langchain/core/documents";
+import { OpenAIEmbeddings } from "@langchain/openai";
+import fs from 'fs';
+import path from 'path';
+import dotenv from 'dotenv';
+
+// Load environment variables from .env file
+dotenv.config({ path: path.resolve(__dirname, '../.env') });
+
+const embeddings = new OpenAIEmbeddings({
+    model: "text-embedding-3-small",
+    apiKey: process.env.OPENAI_API_KEY || ""
+});
+
+const vectorStore = new Chroma(embeddings, {
+    collectionName: "uw-courses-test",
+    url: "http://localhost:8080", // Optional, will default to this value
+    collectionMetadata: {
+      "hnsw:space": "cosine",
+    }, // Optional, can be used to specify the distance method of the embedding space https://docs.trychroma.com/usage-guide#changing-the-distance-function
+  });
+  
+const dataFilePath = path.join(__dirname, '../', 'course-data.json');
+
+async function fetchAndStoreCourseData() {
+    console.log("Fetching course data...");
+    const courseData = await getCourseDataForChroma();
+    console.log("Course data fetched:", courseData);
+
+    fs.writeFileSync(dataFilePath, JSON.stringify(courseData, null, 2));
+    console.log("Course data stored in courseData.json");
+
+    return courseData;
+}
+
+// get course data locally so we don't have to keep pulling from uwaterloo api
+async function getCourseData() {
```

Comment:
So, is the purpose of writing to the file so that we don't unecessarily poll the API? I'm not quite sure how this helps with that if thats the case since we'll have to rerun the whole script anyways whenever we want to get new data.

Reply:
This is just for the setup as it takes a while to get the script going when inititalizing things. We can change this later with ease.

Reply:
ok